src/jit/codegencommon.cpp

```diff
@@ -4888,6 +4888,82 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
 #pragma warning(pop)
 #endif
 
+
+/*****************************************************************************
+* If any Vector3 args are on stack and they are not pass-by-ref, the upper 32bits
+* must be cleared to zeroes. The native compiler doesn't clear the upper bits
+* and there is no way to know if the caller is native or not. So, the upper
+* 32 bits of Vector argument on stack are always cleared to zero.
+*/
+#ifdef FEATURE_SIMD
+void    CodeGen::genClearStackVec3ArgUpperBits()
+{
+#ifdef DEBUG
+    if (verbose)
+        printf("*************** In genClearStackVec3ArgUpperBits()\n");
+#endif
+
+#if !defined(_TARGET_XARCH_)
+    return;
+#endif // !_TARGET_XARCH_
```

Comment:
Since the native compiler doesn't guarantee that the upper 4 bytes are zero initialized, do we have to care about the PInvoke case? It seems to me we really need to care about the RPInvoke only, since the PInvoke is calling native code that has to be able to handle it without problems.